wandb/sdk/wandb_settings.py

```diff
@@ -170,13 +170,18 @@ def _get_program() -> Optional[str]:
         return None
 
 
-def _get_program_relpath_from_gitrepo(
-    program: str, _logger: Optional[_EarlyLogger] = None
+def _get_program_relpath(
+    program: str, root: Optional[str] = None, _logger: Optional[_EarlyLogger] = None
```

Comment:
this refactor allows calling it in system_info, maybe should make it a public method